drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -311,6 +407,190 @@ static void siwx917_ethernet_init(struct net_if *iface)
+	configuration.encryption = SL_WIFI_CCMP_ENCRYPTION;
+	configuration.rate_protocol = SL_WIFI_RATE_PROTOCOL_AUTO;
+	configuration.options = 0;
+	configuration.credential_id = SL_NET_DEFAULT_WIFI_AP_CREDENTIAL_ID;
+	configuration.keepalive_type = SL_SI91X_AP_NULL_BASED_KEEP_ALIVE;
+	configuration.beacon_interval = CONFIG_WIFI_SIWX917_BEACON_INTERVAL;
+	configuration.client_idle_timeout = 0xFF;
+	configuration.dtim_beacon_count = CONFIG_WIFI_SIWX917_DTIM_BEACON_COUNT;
+	configuration.maximum_clients = CONFIG_WIFI_SIWX917_MAX_STATIONS;
+	configuration.beacon_stop = 0;
+	configuration.tdi_flags = SL_WIFI_TDI_NONE;
+	configuration.is_11n_enabled = 1;
+	configuration.ssid.length = params->ssid_length;
+
+	if (params->band != WIFI_FREQ_BAND_2_4_GHZ) {
+		LOG_ERR("Invalid band");
+		return -EINVAL;
+	}
+
+	configuration.channel.band = SL_WIFI_BAND_2_4GHZ;
+
+	switch (params->bandwidth) {
+	case WIFI_FREQ_BANDWIDTH_20MHZ:
+		configuration.channel.bandwidth = SL_WIFI_BANDWIDTH_20MHz;
+		break;
+	case WIFI_FREQ_BANDWIDTH_40MHZ:
+		configuration.channel.bandwidth = SL_WIFI_BANDWIDTH_40MHz;
+		break;
+	case WIFI_FREQ_BANDWIDTH_80MHZ:
+		configuration.channel.bandwidth = SL_WIFI_BANDWIDTH_80MHz;
+		break;
+	default:
+		LOG_ERR("Invalid bandwidth");
+		return -EAGAIN;
+	}
+
+	strncpy(configuration.ssid.value, params->ssid, params->ssid_length);
+
+	switch (params->security) {
+	case WIFI_SECURITY_TYPE_NONE:
+		configuration.security = SL_WIFI_OPEN;
+		break;
+
+	case WIFI_SECURITY_TYPE_PSK:
+		configuration.security = SL_WIFI_WPA2;
+		sl_net_wifi_psk_credential_entry_t wifi_ap_credential = {
+			.type = SL_NET_WIFI_PSK, .data_length = params->psk_length};
+
+		strncpy(wifi_ap_credential.data, params->psk, params->psk_length);
+
+		ret = sl_net_set_credential(SL_NET_DEFAULT_WIFI_AP_CREDENTIAL_ID,
+					    wifi_ap_credential.type, &wifi_ap_credential.data,
+					    wifi_ap_credential.data_length);
+		if (ret) {
+			LOG_ERR("Failed to set credentials: 0x%x", ret);
+			return ERROR_CODE_CHECK(ret);
+		}
+
+		configuration.credential_id = SL_NET_DEFAULT_WIFI_AP_CREDENTIAL_ID;
+		break;
+	default:
+		LOG_ERR("Unsupported security type");
+		return -EINVAL;
+	}
+
+	ret = sl_wifi_start_ap(SL_WIFI_AP_2_4GHZ_INTERFACE, &configuration);
+	if (ret) {
+		LOG_ERR("Failed to enable AP mode: 0x%x", ret);
+		return ERROR_CODE_CHECK(ret);
+	}
+
+	sidev->state = WIFI_STATE_COMPLETED;
+	return ret;
+}
+
+static int siwx917_ap_disable(const struct device *dev)
+{
+	struct siwx917_dev *sidev = dev->data;
+	int ret;
+
+	ret = sl_wifi_stop_ap(SL_WIFI_AP_2_4GHZ_INTERFACE);
+	if (ret) {
+		LOG_ERR("Failed to disable Wi-Fi AP mode: 0x%x", ret);
+		return ERROR_CODE_CHECK(ret);
+	}
+
+	sidev->state = WIFI_STATE_INTERFACE_DISABLED;
+	return ret;
+}
+
+static int siwx917_ap_sta_disconnect(const struct device *dev, const uint8_t *mac_addr)
+{
+	sl_mac_address_t mac = {};
+	int ret;
+
+	if (!mac_addr) {
+		return -EINVAL;
+	}
+
+	strncpy(mac.octet, mac_addr, ARRAY_SIZE(mac.octet));
```

Comment:
From what I can see from this APIs implementation upstream (the documentation itself is a bit unclear) `mac_addr` is *not* a string, so it's invalid and misleading to use `str*()` operations on it. Seems like you should be using `memcpy()` instead. Btw, this might be an opportunity to improve and clarify the upstream API, since AFAIK it's still experimental.